robottelo/hosts.py

```diff
@@ -1392,15 +1394,21 @@ def is_upstream(self):
         :return: True if no downstream satellite RPMS are installed
         :rtype: bool
         """
-        return self.execute('rpm -q satellite-capsule &>/dev/null').status != 0
+        return self.execute(f'rpm -q {self.product_rpm_name} &>/dev/null').status != 0
+
+    @cached_property
+    def is_stream(self):
+        """Check if the Capsule is a stream release or not
+
+        :return: True if the Capsule is a stream release
+        :rtype: bool
+        """
+        return 'stream' in self.execute(f'rpm -q {self.product_rpm_name}').stdout.strip()
 
     @cached_property
     def version(self):
-        if not self.is_upstream:
-            version = self.execute('rpm -q satellite-capsule').stdout
-            return 'stream' if 'stream' in version else version.split('-')[2]
```

Comment:
Please make sure you are not breaking existing tests, especially the tests that spins up dynamic satellite needing snap_version as  `stream` for deploy-satellite WF to deploy the stream satellite.

Reply:
AFAIK, we are not spinning satellites with `deploy_snap_version: stream`. Do we?

Reply:
I mean not the snap version but the satellite version!

Reply:
@ogajduse Have you tested this for stream testing ? Please verify if the stream sat is being deployed for stream testing with this change ?

Reply:
PRT passed